src/server/game/Entities/Player/SocialMgr.cpp

```diff
@@ -125,40 +125,43 @@ void PlayerSocial::SendSocialList(Player* player, uint32 flags)
 {
     ASSERT(player);
 
-    uint32 size = _playerSocialMap.size();
-
     uint32 count = 0;
-    WorldPacket data(SMSG_CONTACT_LIST, (4 + 4 + size * 25)); // just can guess size
-    data << uint32(flags);                                    // 0x1 = Friendlist update. 0x2 = Ignorelist update. 0x4 = Mutelist update.
-    data << uint32(size);                                     // friends count
+
+    WorldPacket data(SMSG_CONTACT_LIST, (4 + 4 + _playerSocialMap.size() * 25)); // just can guess size
+    data << uint32(flags);                                      // 0x1 = Friendlist update. 0x2 = Ignorelist update. 0x4 = Mutelist update.
+
+    size_t wpos = data.wpos();
+    data << uint32(0);                                          // placeholder for contacts count
 
     for (PlayerSocialMap::value_type& v : _playerSocialMap)
     {
-        if (!(v.second.Flags & flags))
+        FriendInfo friendInfo = v.second;
+        if (!(friendInfo.Flags & flags))
             continue;
 
-        ++count;
-        sSocialMgr->GetFriendInfo(player, v.first, v.second);
+        ObjectGuid friendGUID = v.first;
+        sSocialMgr->GetFriendInfo(player, friendGUID, friendInfo);
```

Comment:
You now broke functionality that kept map info updated, because you get the friendInfo in a copy.